vircadia/vircadia-web-sdk
# Switch SDK to TypeScript

- [x] Switch development environment to TypeScript. _Done._
- [x] Convert files to a reasonable level of TypeScript usage. _Done._
- [x] CR, QA, merge.

#### QA

With the domain server running on localhost:
- Enable anonymous connections and restart.

In the directory where the PR files are checked out:
` `npm install` - To set up the the dev & test environment.
- `npm run lint` - Should complete with no errors.
- `npm run build` - Should complete with no errors.
- `npm run test` - All  tests should pass. 
    You can run the tests against the domain server in https://github.com/vircadia/vircadia/pull/1296 .
    Known issue: "Jest did not exit one second after the test run has completed." See: https://github.com/vircadia/vircadia-web-sdk/issues/21
- `npm run devdoc` : Docs should be created in `./docs/dev/` directory.
- `npm run sdkdoc` : Docs should be created in `./docs/sdk/` directory. (Just a couple of placeholder examples at present.)
